cdap-ui/app/cdap/components/PreviewData/index.tsx

```diff
@@ -66,10 +72,22 @@ const PreviewDataViewBase: React.FC<IPreviewDataViewProps> = ({
 
   const [previewLoading, setPreviewLoading] = useState(false);
   const [previewData, setPreviewData] = useState<IPreviewData>({});
+  const [tableData, setTableData] = useState<ITableData>({});
   const [error, setError] = useState(null);
+  const [viewMode, setViewMode] = useState('Table');
+
+  const widgetProps = {
+    on: { value: 'Record', label: 'Record' },
+    off: { value: 'Table', label: 'Table' },
+  };
 
   const updatePreviewCb = (updatedPreview: IPreviewData) => {
     setPreviewData(updatedPreview);
+    const parsedData = getTableData(updatedPreview);
+    setTableData(parsedData);
+    setViewMode(
+      parsedData.inputFieldCount > 100 || parsedData.outputFieldCount > 100 ? 'Record' : 'Table'
```

Comment:
Maybe we can let this number 100 be defined with a constant in case we may need to reuse them

Reply:
Done!